runelite-client/src/main/java/net/runelite/client/util/JoptFileConverter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2020, Trevor <https://github.com/Trevor159>
+ * Copyright (c) 2020, Tomas Slusny <slusnucky@gmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.util;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import joptsimple.ValueConversionException;
+import joptsimple.ValueConverter;
+import net.runelite.client.RuneLite;
+
+public class JoptFileConverter implements ValueConverter<File>
+{
+	@Override
+	public File convert(String fileName)
+	{
+		final File file;
+
+		if (Paths.get(fileName).isAbsolute()
+			|| fileName.startsWith("./")
+			|| fileName.startsWith(".\\"))
+		{
+			file = new File(fileName);
+		}
+		else
+		{
+			file = new File(RuneLite.RUNELITE_DIR, fileName);
+		}
+
+		final Path path = Paths.get(file.getAbsolutePath());
+
+		if (!Files.exists(path) || !Files.isRegularFile(path) || !Files.isWritable(path)) {
```

Comment:
Should the file have to exist? With the current config/session implementation it will generate the file if it does not exist.

Reply:
Oh, right. Yeah i guess, I will change the check